consensus/polybft/sc_integration_test.go

```diff
@@ -7,76 +7,93 @@ import (
 	"strconv"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/abi"
+
 	"github.com/0xPolygon/polygon-edge/chain"
 	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
 	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
 	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi/artifact"
 	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
 	"github.com/0xPolygon/polygon-edge/consensus/polybft/wallet"
 	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/crypto"
 	secretsHelper "github.com/0xPolygon/polygon-edge/secrets/helper"
 	"github.com/0xPolygon/polygon-edge/state"
 	"github.com/0xPolygon/polygon-edge/types"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-	"github.com/umbracle/ethgo"
-	"github.com/umbracle/ethgo/abi"
 )
 
 func TestIntegratoin_PerformExit(t *testing.T) {
 	t.Parallel()
 
-	// create validator set
+	const gasLimit = 1000000000000
+
+	// create validator set and checkpoint mngr
 	currentValidators := newTestValidatorsWithAliases(t, []string{"A", "B", "C", "D"}, []uint64{100, 100, 100, 100})
 	accSet := currentValidators.getPublicIdentities()
+	cm := checkpointManager{blockchain: &blockchainMock{}}
 
 	senderAddress := types.Address{1}
 	bn256Addr := types.Address{2}
-	l1StateReceiverAddr := types.Address{3}
+	stateSenderAddr := types.Address{5}
+	receiverAddr := types.Address{6}
+	amount1 := big.NewInt(3)
+	amount2 := big.NewInt(2)
 
 	alloc := map[types.Address]*chain.GenesisAccount{
-		senderAddress:         {Balance: big.NewInt(100000000000)},
+		senderAddress:         {Balance: ethgo.Ether(100)},
 		contracts.BLSContract: {Code: contractsapi.BLS.DeployedBytecode},
 		bn256Addr:             {Code: contractsapi.BLS256.DeployedBytecode},
-		l1StateReceiverAddr:   {Code: contractsapi.TestL1StateReceiver.DeployedBytecode},
+		stateSenderAddr:       {Code: contractsapi.StateSender.DeployedBytecode},
 	}
 	transition := newTestTransition(t, alloc)
 
 	getField := func(addr types.Address, abi *abi.ABI, function string, args ...interface{}) []byte {
 		input, err := abi.GetMethod(function).Encode(args)
 		require.NoError(t, err)
 
-		result := transition.Call2(senderAddress, addr, input, big.NewInt(0), 1000000000)
-		require.NoError(t, result.Err)
+		result := transition.Call2(senderAddress, addr, input, big.NewInt(0), gasLimit)
```

Comment:
I know that is not a part of this PR, but why don't we submit transition as a parameter of  `getField := func(addr types.Address...` ?

Reply:
`senderAddress` as well

Reply:
can be done later